### ISSUE TITLE
Optimizer='auto' problem

### ISSUE THREAD

**Issue by Jasonfengjx** [ultralytics/ultralytics#14527]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When I use auto, the training prcess is normal.But when I change the optimizer(same parameter) the training will be abnormal.
Auto:optimizer: 'optimizer=auto' found, ignoring 'lr0=0.03' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 64 weight(decay=0.0), 91 weight(decay=0.004), 131 bias(decay=0.0)

Change:

optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 64 weight(decay=0.0), 91 weight(decay=0.004), 131 bias(decay=0.0)

results are different.

**Comment by Jasonfengjx** [comment]
Why will this phenomenon occur?

**Comment by pderrenger** [comment]
@Jasonfengjx the difference in results when switching from `optimizer='auto'` to manually setting the optimizer is likely due to the automatic adjustment of hyperparameters like learning rate (`lr0`) and momentum when using the 'auto' setting. The 'auto' optimizer dynamically determines the best settings for your specific training scenario, which can lead to different training dynamics compared to manually specified values. If you encounter issues with manual settings, ensure that the learning rate and momentum are appropriately tuned for your dataset and model. Additionally, please verify that you are using the latest version of the Ultralytics package to rule out any potential bugs.

**Comment by karolbadowski** [comment]
I would like to use OneCycle instead of auto, with lr0 that works the best for my organisation and for our custom training anf finetunng strategy.

In the past OneCycle helped us reach much better results than current Auto.

Auto causes huge drops of mAP50-95 in epovh 4 (even down to 0.

How can we use simple OneCycle with predefined lr0?
It seems to not be on the list of the optomizers to choose from.

<img width="1013" height="295" alt="Image" src="https://github.com/user-attachments/assets/a9bdcc1c-35c2-42c4-b254-73708802a596" />

**Comment by glenn-jocher** [comment]
@karolbadowski OneCycle isn’t an optimizer; it’s the LR schedule—keep your preferred optimizer and use OneCycle by leaving `cos_lr=False` (default), set your `lr0`, and optionally `lrf` to control the final LR fraction; details in the Train settings and the `one_cycle()` reference. ([docs.ultralytics.com](https://docs.ultralytics.com/modes/train/))

```bash
yolo detect train model=yolo11n.pt data=coco8.yaml epochs=100 optimizer=AdamW lr0=0.003 lrf=0.01 momentum=0.9 cos_lr=False
```

For parameter references, see the Train page (optimizer, `lr0`, `lrf`, `cos_lr`) and the `one_cycle()` scheduler helper. If you’d like to continue this on a closed/stale thread, please open a new issue for visibility. [Model Training settings](https://docs.ultralytics.com/modes/train/) | [`one_cycle()` reference](https://docs.ultralytics.com/reference/utils/torch_utils/#one_cycle)
